member_names.go

```diff
@@ -86,9 +86,29 @@ func validateMapMemberNames(m map[string]any, mode MemberNameValidationMode) err
 }
 
 func validateJSONMemberNames(b []byte, mode MemberNameValidationMode) error {
+	// do not unmarshal if validation is disabled
+	if mode == DisableValidation {
+		return nil
+	}
+
 	var m map[string]any
 	if err := json.Unmarshal(b, &m); err != nil {
 		return fmt.Errorf("unexpected unmarshal failure: %w", err)
 	}
 	return validateMapMemberNames(m, mode)
 }
+
+// func validateJSONMemberNames2(b []byte, mode MemberNameValidationMode) error {
```

Comment:
leftover artifact of testing this out?

Reply:
OOPS yes sorry. I try to review my changes before sending them but clearly was sloppy here thanks.